operators/argo-controller/tests/integration/test_charm.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+log = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+CHARM_ROOT = "."
+APP_NAME = "argo-controller"
+
+MINIO_CONFIG = {
+    "access-key": "minio",
+    "secret-key": "minio-secret-key",
+}
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy_with_relations(ops_test: OpsTest):
+    built_charm_path = await ops_test.build_charm(CHARM_ROOT)
+    log.info(f"Built charm {built_charm_path}")
+
+    image_path = METADATA["resources"]["oci-image"]["upstream-source"]
+    resources = {"oci-image": image_path}
+
+    await ops_test.model.deploy(
+        entity_url=built_charm_path,
+        application_name=APP_NAME,
+        resources=resources,
+    )
+
+    # Deploy required relations
+    await ops_test.model.deploy(entity_url="minio", config=MINIO_CONFIG)
+    await ops_test.model.add_relation(
+        f"{APP_NAME}:object-storage", "minio:object-storage"
+    )
+
+    await ops_test.model.wait_for_idle(timeout=60 * 10)
+    # TODO: This does not handle blocked status right.  Sometimes it passes when argo-controller
```

Comment:
This was one that gave me intermittent issues but **usually** worked (and whenever it failed, it was always a false failure rather than letting the test pass when it shouldn't).  My thought was to get these tests merged, bump the argo-controller's container, and then iterate on this afterwards since we have the kf 1.4 deadline looming.  I'm flexible though

Reply:
Sure, iterating on this sounds good